packages/shared/lib/services/connection.service.ts

```diff
@@ -1451,6 +1544,67 @@ class ConnectionService {
         }
     }
 
+    public getJwtCredentials(
+        provider: ProviderJwt,
+        privateKeyId?: string,
+        privateKey?: string,
+        issuerId?: string,
+        apiKey?: string
+    ): ServiceResponse<JWTCredentials> {
+        if (apiKey) {
+            const apiKeyParts = apiKey.split(':');
+            if (apiKeyParts.length === 2 && apiKeyParts[0] && apiKeyParts[1]) {
+                privateKeyId = apiKeyParts[0];
+                privateKey = apiKeyParts[1];
+            } else {
+                throw new NangoError(`invalid_api_key_format`);
+            }
+        }
+
+        if (!privateKey) {
+            throw new NangoError(`invalid_jwt_private_key`);
+        }
+
+        const now = Math.floor(Date.now() / 1000);
+        const payload = {
+            iat: now,
+            exp: now + provider.token.expires_in_ms / 1000,
+            aud: provider.token.payload.aud
```

Comment:
can we do?
```
const payload = {
            ...provider.token.payload.
            iat: now,
            exp: now + provider.token.expires_in_ms / 1000
```
so any values defined in providers.yaml will be set automatically

Reply:
yes, updated this with the latest commit.